buffer/src/main/java/io/netty/buffer/SimpleLeakAwareCompositeByteBuf.java

```diff
@@ -16,64 +16,86 @@
 package io.netty.buffer;
 
 
-import io.netty.util.ResourceLeak;
+import io.netty.util.ResourceLeakTracker;
+import io.netty.util.internal.ObjectUtil;
 
 import java.nio.ByteOrder;
 
-final class SimpleLeakAwareCompositeByteBuf extends WrappedCompositeByteBuf {
+class SimpleLeakAwareCompositeByteBuf extends WrappedCompositeByteBuf {
 
-    private final ResourceLeak leak;
+    final ResourceLeakTracker<ByteBuf> leak;
 
-    SimpleLeakAwareCompositeByteBuf(CompositeByteBuf wrapped, ResourceLeak leak) {
+    SimpleLeakAwareCompositeByteBuf(CompositeByteBuf wrapped, ResourceLeakTracker<ByteBuf> leak) {
         super(wrapped);
-        this.leak = leak;
+        this.leak = ObjectUtil.checkNotNull(leak, "leak");
     }
 
     @Override
-    public boolean release() {
-        boolean deallocated = super.release();
-        if (deallocated) {
-            leak.close();
+    public final boolean release() {
+        // Call unwrap() before just in case that super.release() will change the ByteBuf instance that is returned
+        // by unwrap().
+        ByteBuf unwrapped = unwrap();
+        if (super.release()) {
+            closeLeak(unwrapped);
+            return true;
         }
-        return deallocated;
+        return false;
     }
 
     @Override
-    public boolean release(int decrement) {
-        boolean deallocated = super.release(decrement);
-        if (deallocated) {
-            leak.close();
+    public final boolean release(int decrement) {
+        // Call unwrap() before just in case that super.release() will change the ByteBuf instance that is returned
+        // by unwrap().
+        ByteBuf unwrapped = unwrap();
+        if (super.release(decrement)) {
+            closeLeak(unwrapped);
+            return true;
         }
-        return deallocated;
+        return false;
+    }
+
+    private void closeLeak(ByteBuf trackedByteBuf) {
+        // Close the ResourceLeakTracker with the tracked ByteBuf as argument. This must be the same that was used when
+        // calling DefaultResourceLeak.track(...).
+        boolean closed = leak.close(trackedByteBuf);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this useless assignment to local variable "closed". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1854)